listenbrainz/db/fresh_releases.py

```diff
@@ -0,0 +1,94 @@
+                                      , array_agg(distinct a.gid) AS artist_mbids
+                                      , rgpt.name AS release_group_primary_type
+                                      , rgst.name AS release_group_secondary_type
+                                      , row_number() OVER (PARTITION BY rg.id ORDER BY make_date(rgm.first_release_date_year,
+                                                                                                 rgm.first_release_date_month,
+                                                                                                 rgm.first_release_date_day)
+                                                                                                ) AS rnum
+                                  FROM release rl
+                                  JOIN release_group rg
+                                    ON rl.release_group = rg.id
+                                  JOIN release_group_meta rgm
+                                    ON rgm.id = rg.id
+                             LEFT JOIN release_group_primary_type rgpt
+                                    ON rg.type = rgpt.id
+                             LEFT JOIN release_group_secondary_type_join rgstj
+                                    ON rgstj.release_group = rg.id
+                             LEFT JOIN release_group_secondary_type rgst
+                                    ON rgstj.secondary_type = rgst.id
+                                  JOIN artist_credit ac
+                                    ON rl.artist_credit = ac.id
+                                  JOIN artist_credit_name acn
+                                    ON acn.artist_credit = ac.id
+                                  JOIN artist a
+                                    ON acn.artist = a.id
+                                 WHERE make_date(rgm.first_release_date_year,
+                                                 rgm.first_release_date_month,
+                                                 rgm.first_release_date_day) >= %s
+                                   AND make_date(rgm.first_release_date_year,
+                                                 rgm.first_release_date_month,
+                                                 rgm.first_release_date_day) <= %s
+                              GROUP BY rg.id
+                                     , release_date
+                                     , release_mbid
+                                     , release_name
+                                     , release_date
+                                     , artist_credit_name
+                                     , release_group_primary_type
+                                     , release_group_secondary_type
+                        ) AS q
+                  WHERE q.rnum = 1
+               ORDER BY release_date
+                      , artist_credit_name
+                      , release_name"""
+
+    psycopg2.extras.register_uuid()
```

Comment:
Probably register somewhere in app or database engine creation?

Reply:
Done.